File: app/src/main/java/com/example/activity/MainActivity.java
package com.example.activity;

import android.content.Intent;
import android.os.Bundle;
import android.widget.Button;

import com.example.mydemo.R;

import androidx.appcompat.app.AppCompatActivity;
import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.OnClick;

public class MainActivity extends AppCompatActivity {

	@BindView(R.id.btn_algorithm)
	Button btnAlgorithm;

	@Override
	protected void onCreate(Bundle savedInstanceState) {
		super.onCreate(savedInstanceState);
		setContentView(R.layout.activity_main);
		ButterKnife.bind(this);
	}


	@OnClick(R.id.btn_algorithm)
	public void onViewClicked() {
		startActivity(new Intent(MainActivity.this,AlgorithmActivity.class));
	}
}

File: app/src/main/java/com/example/activity/MainActivity.kt
package com.example.activity

import android.os.Bundle
import android.os.PersistableBundle
import androidx.appcompat.app.AppCompatActivity
import com.example.mydemo.databinding.ActivityMainBinding

/**
 *@package com.example.activity
 *@author https://github.com/asd3590058
 *@fileName MainActivity
 *@date 2022/11/8 11:13
 *@description
 */
class MainActivity : AppCompatActivity() {
    private val binding by lazy { ActivityMainBinding.inflate(layoutInflater) }
    override fun onCreate(savedInstanceState: Bundle?, persistentState: PersistableBundle?) {
        super.onCreate(savedInstanceState, persistentState)
        setContentView(binding.root)
    }
}